src/server/pfs/server/server_test.go

```diff
@@ -575,6 +575,30 @@ func TestInspectFile(t *testing.T) {
 	require.Equal(t, pfsclient.FileType_FILE_TYPE_REGULAR, fileInfo.FileType)
 	require.Equal(t, len(fileContent1), int(fileInfo.SizeBytes))
 
+	// We inspect the file with three filter shards that have different block
+	// numbers, so that only one of the filter shards should match the file
+	// since the file only contains one block.  However, the way PFS is currently
+	// implemented, one of the shards might see an empty version of the file.
+	_, err1 := client.InspectFile(repo, commit1.ID, "foo", "", &pfsclient.Shard{
+		BlockNumber:  0,
+		BlockModulus: 3,
+	})
+	_, err2 := client.InspectFile(repo, commit1.ID, "foo", "", &pfsclient.Shard{
+		BlockNumber:  1,
+		BlockModulus: 3,
+	})
+	_, err3 := client.InspectFile(repo, commit1.ID, "foo", "", &pfsclient.Shard{
+		BlockNumber:  2,
+		BlockModulus: 3,
+	})
+	var numNotFound int
+	for _, err := range []error{err1, err2, err3} {
+		if err != nil && strings.Contains(err.Error(), "not found") {
+			numNotFound++
+		}
+	}
+	require.True(t, numNotFound == 1 || numNotFound == 2)
```

Comment:
This seems like a kind of weird behavior to me that one container sees an empty version of the file. I was thinking that you'd only see the empty file if there were no blocks at all the show to any of the containers.